docs/quick_tutorial/authentication/tutorial/security.py

```diff
@@ -1,5 +1,21 @@
-USERS = {'editor': 'editor',
-         'viewer': 'viewer'}
+import os
+import hashlib
+import binascii
+
+SALT_LENGTH = 16
+
+def hash_password_with_salt(password, salt=None):
+    salt = salt or os.urandom(SALT_LENGTH)
+    dk = hashlib.pbkdf2_hmac('sha512', password.encode('utf-8'), salt, 100000)
+    return binascii.hexlify(dk + salt).decode('utf-8')
```

Comment:
@bertjwregeer can you look this over and provide feedback?

This does not look right to me, because the function `pbkdf2_hmac` in the preceding line already hashes the concatenation of the password and salt behind the scenes.

Also 100000 iterations was OK in 2013 for SHA-256, but we're now in 2016. I don't know whether it needs to be adjusted for time as well as for SHA-512. I'd suggest using a variable of `iterations` to be consistent with the function's argument name.

Also we want to avoid the en/decode utf-8 stuff, as referred in issue #2623.

This goes for authorization as well.